### ISSUE TITLE
Could not reproduce the results

### ISSUE THREAD

**Issue by kuntalkumarpal** [ShannonAI/mrc-for-flat-nested-ner#9]
Hi, 
I was trying to run the code for **ace2005** and I am not able to reproduce the results with the preprocessed data that have been added to the data-preprocess section

* I noticed if I allow it to evaluate every step that is bypassing the following line in train_bert_mrc.py file then the code executes
`if nb_tr_steps % config.checkpoint == 0:`
* Also there seems to be another issue in function **bmes_decode** in file **metric/flat_span_f1.py** in line _152_, if the current_label =='E' without a preceding 'B' then it goes to an infinite loop. Shouldn't there be a `idx +=1` in line 152 in the else condition?

If I do these changes then I code is getting evaluated but the results are not reproduced. Can you please verify the code one and also comment if there is any issue in the command I am running

`FOLDER_PATH=<mypath>/mrc-for-flat-nested-ner
CONFIG_PATH=<mypath>/mrc-for-flat-nested-ner/config/en_bert_base_cased.json
DATA_PATH=<mypath>/release/en_ace05/
EXPORT_DIR=<mypath>/release/en_ace05/output
data_sign=ace2005
entity_sign=flat`

export PYTHONPATH=${FOLDER_PATH}
python ${FOLDER_PATH}/run/train_bert_mrc.py \
--config_path ${CONFIG_PATH} \
--data_dir ${DATA_PATH} \
--bert_model bert-base-cased \
--output_dir ${EXPORT_DIR} \
--entity_sign ${entity_sign} \
--data_sign ${data_sign} \
--export_model True \
--dropout 0.3 \
--checkpoint 1000 \
--max_seq_length 128 \
--train_batch_size 16 \
--dev_batch_size 16 \
--test_batch_size 16 \
--learning_rate 5e-5 \
--weight_start 1.0 \
--weight_end 1.0 \
--weight_span 1.0 \
--num_train_epochs 2 \
--seed 42 \
--n_gpu 1 \
--warmup_proportion -1 \
--gradient_accumulation_steps 1`

I got the following results

`-*--*--*--*--*--*--*--*--*--*-
current data_sign: ace2005
=*==*==*==*==*==*==*==*==*==*=
loading train data ... ...
50995
50995 train data loaded
=*==*==*==*==*==*==*==*==*==*=
loading dev data ... ...
6776
6776 dev data loaded
=*==*==*==*==*==*==*==*==*==*=
loading test data ... ...
7406
7406 test data loaded
######################################################################
EPOCH:  0
-*--*--*--*--*--*--*--*--*--*--*--*--*--*--*-
current training loss is :
2.0820212364196777
............................................................
DEV: loss, acc, precision, recall, f1
1.5432 0.0782 0.0006 0.0007 0.0006
............................................................
TEST: loss, acc, precision, recall, f1
1.5449 0.0757 0.0011 0.0018 0.0014
-*--*--*--*--*--*--*--*--*--*--*--*--*--*--*-
-*--*--*--*--*--*--*--*--*--*--*--*--*--*--*-
current training loss is :
1.5602871179580688
............................................................
DEV: loss, acc, precision, recall, f1
1.0879 0.7018 0.0 0.0 0.0
-*--*--*--*--*--*--*--*--*--*--*--*--*--*--*-
-*--*--*--*--*--*--*--*--*--*--*--*--*--*--*-
current training loss is :
1.0668576955795288
............................................................
DEV: loss, acc, precision, recall, f1
0.7798 0.7634 0.0 0.0 0.0
-*--*--*--*--*--*--*--*--*--*--*--*--*--*--*-
-*--*--*--*--*--*--*--*--*--*--*--*--*--*--*-
current training loss is :
0.7793314456939697
............................................................
DEV: loss, acc, precision, recall, f1
0.5376 0.7644 0.0 0.0 0.0
-*--*--*--*--*--*--*--*--*--*--*--*--*--*--*-
-*--*--*--*--*--*--*--*--*--*--*--*--*--*--*-
current training loss is :
0.6774319410324097
............................................................
DEV: loss, acc, precision, recall, f1
0.3504 0.7644 0.0 0.0 0.0
-*--*--*--*--*--*--*--*--*--*--*--*--*--*--*-
-*--*--*--*--*--*--*--*--*--*--*--*--*--*--*-
current training loss is :
0.3500059247016907`

I also ran with **entity_sign=nested**. Even that does not work.
Can you please see if I am missing anything

**Comment by littlesulley** [comment]
ace2005 is a flat-NER dataset. 
Please follow the log here 
https://github.com/ShannonAI/mrc-for-flat-nested-ner/blob/master/log/en_ace05.txt
to reproduce the performance.
